build.gradle

```diff
@@ -406,10 +407,92 @@ configure(project(':core')) {
         systemProperty 'jdk.attach.allowAttachSelf', true
     }
 
-    task havenoDeps(type: Exec) {
-        workingDir '..'
-        commandLine './scripts/haveno_deps.sh'
+    task havenoDeps {
+        doLast {
+            // get monero binaries download url
+            Map moneroBinaries = [
+                'linux'         : 'https://github.com/haveno-dex/monero/releases/download/testing8/monero-bins-haveno-linux.tar.gz',
+                'linux-sha256'  : 'a222bdae08e5464d1b751e6301144a904578bef37c75444c46aef762299c0b92',
+                'mac'           : 'https://github.com/haveno-dex/monero/releases/download/testing8/monero-bins-haveno-mac.tar.gz',
+                'mac-sha256'    : 'fbebd9eb40550cd992bf9e1f946741d5adc564c428b9d4c78981363235c4be89',
+                'windows'       : 'https://github.com/haveno-dex/monero/releases/download/testing8/monero-bins-haveno-windows.zip',
+                'windows-sha256': 'c9e6150ba9cb8b35b7c63ac8786610e3c37725a973e35c09e6fbbfe20430302a'
+            ]
+
+            String osKey
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                osKey = 'windows'
+            } else if (Os.isFamily(Os.FAMILY_MAC)) {
+                osKey = 'mac'
+            } else {
+                osKey = 'linux'
+            }
+            String moneroDownloadUrl = moneroBinaries[osKey]
+            String moneroSHA256Hash = moneroBinaries[osKey + '-sha256']
+            String moneroArchiveFileName = moneroDownloadUrl.tokenize('/').last()
+
+            // download
+            String localnetDirName = '.localnet'
+            File localnetDir = new File(project.rootDir, localnetDirName)
+            localnetDir.mkdirs()
+            File moneroArchiveFile = new File(localnetDir, moneroArchiveFileName)
+            if (!moneroArchiveFile.exists()) {
+                ext.downloadAndVerifyArchive(moneroDownloadUrl, moneroSHA256Hash, moneroArchiveFile)
+            }
+
+            // extract
+            String monerod = 'monerod'
+            String moneroRpc = 'monero-wallet-rpc'
+            if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+                monerod = monerod + '.exe'
+                moneroRpc = moneroRpc + '.exe'
+                ext.extractArchiveZip(moneroArchiveFile, localnetDir)
```

Comment:
We should not extract the archive files if the archive already existed with the correct hash, and the binaries already exist. This allows developers to easily test new binaries without being overwritten on build, until the next binary release when the binaries will be automatically updated. This would match current functionality from the build script.

Reply:
Implementing.